src/drive/web/modules/queries.js

```diff
@@ -42,4 +47,25 @@ const buildQuery = ({ currentFolderId, type, sortAttribute, sortOrder }) => ({
   }
 })
 
+/**
+ * Get the query for folder if given the query for files
+ * and vice versa.
+ *
+ * If given the queryId `directory id123 name desc`, will return
+ * the query `files id123 name desc`.
+ */
+export const getMirrorQueryId = queryId => {
+  const { type, folderId, sortAttribute, sortOrder } = parseFolderQueryId(
+    queryId
+  )
+  const otherType = type === 'directory' ? 'file' : 'directory'
+  const otherQueryId = formatFolderQueryId(
+    otherType,
+    folderId,
+    sortAttribute,
+    sortOrder
+  )
+  return otherQueryId
+}
```

Comment:
Are you considering that the query `directory 123 name desc` returns the same values than `files 123 name desc`? How is that even possible to have the both queries?

Reply:
No, those are two different queries, one resolving with the directories of the folder, the other with the files.